### ISSUE TITLE
Allow drag and drop reordering in the same animation group.

### ISSUE THREAD

**Issue by rocamocha** [defold/defold#5569]
**Is your feature request related to a problem? Please describe (REQUIRED):**
I've been finding Defold very wonderful to work with so far, but there are a two snags regarding how the editor handles and organizes assets (images in particular) that I've tripped on enough times to make me write this. Currently, when you have an animation group in an atlas, it is **only** possible to reorder the images in the group by cut + paste or moving between **separate** animation groups. This means that every image of the set must always be correctly ordered by the editor's sorting in order to add them all at once, otherwise you will be making multiple trips to the "add images" dialogue box or cutting and pasting multiple times.

Now this alone wouldn't be too bad, it just forces proper asset management for developers - but the real problem emerges when you take into consideration that the editor seems to sort assets by name **lexicographically** which means that if your numbering does not contain leading zeroes, the assets will be sorted in the following order:

1, 10, 11, 12, 13, 14, 15, 2, 3, 4, 5, 6, 7, 8, 9

...where 5 is greater than 11 because 11 starts with a 1.

**Describe the solution you'd like (REQUIRED):**
Either of two solutions would solve this problem:
1. Allow us to reorder images in animation groups by drag and drop.
2. Make the sorting method respect numbers as integers and have filenames numbered like "idle14" sorted correctly.

**Describe alternatives you've considered (REQUIRED):**
Batch renaming can be tedious, especially when assets are being iterated upon consistently, and some programs like Aseprite simply aren't capable of exporting in numbering systems with leading zeroes.

![no-drag-and-drop](https://user-images.githubusercontent.com/30216269/107950431-c93df800-6f3a-11eb-9364-b63f0a97f74b.gif)

**Comment by britzl** [comment]
> it is **only** possible to reorder the images in the group by cut + paste or moving between **separate** animation groups

I agree that we should add drag'n'drop support, but I wanted to point out that you can change the order of animations through alt and + or - (it's in the menu as Move Up and Move Down I think).

**Comment by rocamocha** [comment]
Any chance on the sorting part of the post?


> > it is **only** possible to reorder the images in the group by cut + paste or moving between **separate** animation groups
> 
> I agree that we should add drag'n'drop support, but I wanted to point out that you can change the order of animations through alt and + or - (it's in the menu as Move Up and Move Down I think).

Ah, that's good to know. I've just tested it out though, and the feature is not there when trying to do it with multiple assets selected, which is the big issue. So in the case with the sorting method mentioned with more than 9 assets, you'd have to be moving them one by one to use that method.

**Comment by beldenfox** [comment]
I just ran into this myself and I vote for fixing this by altering the initial sorting. The visual file navigators in Windows 10, macOS, and Ubuntu all use a mix of lexicographic and numeric sorting to provide a more logical sort order than the Defold editor does. Updating the initial sorting in the editor to match the platforms would probably fix this problem in 99% of the cases.
